service/storage_service.cc

```diff
@@ -154,6 +154,14 @@ namespace {
 
 static constexpr std::chrono::seconds wait_for_live_nodes_timeout{30};
 
+static future<> run_on_repair_service(sharded<repair_service>& repair, std::function<future<>(repair_service&)> f) {
+    if (!repair.local_is_initialized()) {
+        throw seastar::abort_requested_exception();
```

Comment:
Why throw this instead of gate_closed, or just leave the gate to throw it?

Reply:
We can't rely on the gate as `repair_service` might have been stopped at this point so local instances would be destroyed.

`repair_service::stop` would be called on shutdown, that's why I thought `abort_requested_exception` would fit

Reply:
I see, makes sense.

I think a better name for this method would be: `do_with_repair_service()`.